### ISSUE TITLE
Show an error when using a name prior to usage of global keyword

### ISSUE THREAD

**Issue by Jackenmen** [charliermarsh/ruff#197]
e.g. this will error at runtime:
```py
TOMATO = "black cherry"


def update_tomato():
    print(TOMATO)  # will error here
    global TOMATO
    TOMATO = "cherry tomato"
```

While this won't:
```py
TOMATO = "black cherry"


def update_tomato():
    global TOMATO
    print(TOMATO)
    TOMATO = "cherry tomato"
```

Seen in pylint:
https://pylint.pycqa.org/en/latest/user_guide/messages/error/used-prior-global-declaration.html

**Comment by charliermarsh** [comment]
This is also [E0118](https://pylint.pycqa.org/en/latest/user_guide/messages/error/used-prior-global-declaration.html) in Pylint.

**Comment by Jackenmen** [comment]
I think I may have originally wanted this to be added to `F821` (`UndefinedName`) but I guess it isn't quite the same kind of error + flake8 doesn't report on it:
```
  File "/tmp/blah/test.py", line 6
    global TOMATO
    ^^^^^^^^^^^^^
SyntaxError: name 'TOMATO' is used prior to global declaration
```

**Comment by charliermarsh** [comment]
Gonna try to fix this today. (It’ll force me to learn all the rules around globals.)

**Comment by charliermarsh** [comment]
Interesting I think these are actually _syntax_ errors?

**Comment by charliermarsh** [comment]
Implemented as `PLE0118` (to match Pylint's `E0118`).
